d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadStateManager.java

```diff
@@ -60,17 +60,24 @@ public class DualReadStateManager
 
   private final DualReadLoadBalancerJmx _dualReadLoadBalancerJmx;
 
-  private final DualReadLoadBalancerMonitor.UriPropertiesDualReadMonitor _uriPropertiesDualReadMonitor;
   private final DualReadLoadBalancerMonitor.ServicePropertiesDualReadMonitor _servicePropertiesDualReadMonitor;
   private final DualReadLoadBalancerMonitor.ClusterPropertiesDualReadMonitor _clusterPropertiesDualReadMonitor;
+  private final UriPropertiesDualReadMonitor _uriPropertiesDualReadMonitor;
+  private final boolean _monitorUriProperties;
 
-
+  @Deprecated
   public DualReadStateManager(DualReadModeProvider dualReadModeProvider, ScheduledExecutorService executorService)
+  {
+    this(dualReadModeProvider, executorService, false);
```

Comment:
Should this default to `true`? I guess we still need to benchmark it

Reply:
right, that's why the default should be disabled (false), right? so that we can enabled it on some app only for testing.

Reply:
Yeah makes sense